x/transferinject/keeper/keeper.go

```diff
@@ -0,0 +1,168 @@
+package keeper
+
+import (
+	"errors"
+	"fmt"
+	. "slices"
+
+	errorsmod "cosmossdk.io/errors"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	errortypes "github.com/cosmos/cosmos-sdk/types/errors"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	transfertypes "github.com/cosmos/ibc-go/v6/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v6/modules/core/04-channel/types"
+	porttypes "github.com/cosmos/ibc-go/v6/modules/core/05-port/types"
+
+	rtypes "github.com/dymensionxyz/dymension/v3/x/rollapp/types"
+	"github.com/dymensionxyz/dymension/v3/x/transferinject/types"
+)
+
+type Keeper struct {
+	cdc codec.BinaryCodec
+	porttypes.IBCModule
+	porttypes.ICS4Wrapper
+	rollappKeeper types.RollappKeeper
+	bankKeeper    types.BankKeeper
+}
```

Comment:
With the chosen protoco, it is possible to send the denom metadata UNTIL the ack comes back

which migh result in duplicated sends

Can you document this somewhere in a comment and state that we're ok with that


(cc @omritoptix I guess we're ok with it right?)